### PR TITLE
Gnome shell 3.36.1

### PR DESCRIPTION
**Pull request by hedning**
###### Motivation for this change

Bump gnome-shell. This unfortunately breaks the extension app, as they've introduced a new internal library ([`libshew`](https://gitlab.gnome.org/GNOME/gnome-shell/-/tree/master/subprojects%2Fshew)) which gets an incorrect path in its typelib ala. eog.

Not sure if we've found a good way to deal with these failures (I seem to recall we've fixed something like this before, but can't quite remember). 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome-3/core/gnome-shell/default.nix [hunk 1]
```diff
@@ -48,51 +50,19 @@ in stdenv.mkDerivation rec {
...
-      sha256 = "RBA+JHz4ZvmbJZMnGNieD6D5LONRgFU4iOFIMQQ2kHQ=";
-    })
-
-    # Fix Telepathy chat integration.
-    # https://gitlab.gnome.org/GNOME/gnome-shell/issues/2449
-    (fetchpatch {
-      url = "https://gitlab.gnome.org/GNOME/gnome-shell/commit/766288eec1bd3bd50dfc4ddf410c2b507187e603.patch";
-      sha256 = "Cp6xLohCM0gmMxtyYjSukS2oV60Khmxf4iQd9EDAlIc=";
-    })
+    ./fix-bash-completion.patch
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/core/gnome-shell/default.nix:65]: Do you have link to an upstream MR?

**Comment by jtojnar** [comment]
Our gobject-introspection patches require that the [`install_dir` for the library](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/7d7a15f978c7ff27476e984ab21e5f00a1922c5c/subprojects/shew/src/meson.build#L16) in meson is absolute. So replace it with `get_option('prefix') / pkglibdir`.

File: pkgs/desktops/gnome-3/core/gnome-shell/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/desktops/gnome-3/core/gnome-shell/default.nix:65]: I did it here https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1195

**Comment by worldofpeace** [comment]
Upstream patch for the define variable thing https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1195

File: pkgs/desktops/gnome-3/core/gnome-shell/default.nix [hunk 2]
```diff
@@ -48,51 +50,26 @@ in stdenv.mkDerivation rec {
...
-
-    # Fix Telepathy chat integration.
-    # https://gitlab.gnome.org/GNOME/gnome-shell/issues/2449
-    (fetchpatch {
-      url = "https://gitlab.gnome.org/GNOME/gnome-shell/commit/766288eec1bd3bd50dfc4ddf410c2b507187e603.patch";
-      sha256 = "Cp6xLohCM0gmMxtyYjSukS2oV60Khmxf4iQd9EDAlIc=";
-    })
+    # Use absolute path for libshew installation to make our patched gobject-introspection
+    # aware of the location to hardcode in the generated GIR file.
+    ./shew-gir-path.patch
```
Comment (worldofpeace) [inline pkgs/desktops/gnome-3/core/gnome-shell/default.nix:72]: I think you forgot to commit this.

**Comment by worldofpeace** [comment]
We have a request to add another patch https://github.com/NixOS/nixpkgs/issues/84805.
This one in mutter https://gitlab.gnome.org/GNOME/mutter/-/commit/23d0bdd46d849379ed4c0146fbc1784f5f32f2f8.

**Comment by jtojnar** [comment]
That commit does not look like anything but cleanup.

**Comment by worldofpeace** [comment]
Oops, I think I grabbed the wrong commit. See https://gitlab.gnome.org/GNOME/mutter/-/issues/1128

**Comment by jtojnar** [comment]
That should already be included: https://gitlab.gnome.org/GNOME/mutter/-/commit/92710d8f895188ffee99a50079c96f2a2354eebd

**Comment by hedning** [comment]
Ok, two remaining issues. Looks like the extension app is now activated through dbus or something, so the wrapper is no longer effective. Adding gtk to the search path fixed the issue.

Also forgot about this, but the size of the nixos-graphics in gdm is now wrong for some reason:
image

**Comment by jtojnar** [comment]
I am running this now and did not notice anything other than:

Screenshot from 2020-04-10 09-58-00

Reverting gnome-hell update fixes it.

**Comment by hedning** [comment]
Yeah, looks like this is the problem with the vendor logo: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2591, they use the «natural» scale of the image, which in our case is rather large.

**Comment by hedning** [comment]
Hmm, right, I'm not actually able launch the extension app, getting this error:
```
april 10 11:37:07 x230 gjs[6098]: JS ERROR: Error: Requiring Gtk, version 3.0: Typelib file for namespace 'Gtk', version '3.0' not found
                                  @resource:///org/gnome/Shell/Extensions/js/main.js:6:17
                                  start@resource:///org/gnome/gjs/modules/script/package.js:206:5
                                  @/nix/store/v4cqz1r08gfvhfyrxq957di3f7iiqfwp-gnome-shell-3.36.1/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```
I tested things in a nested wayland shell, and that happened to fix the issue (probably seeded the environment with the correct typelib paths).

@jtojnar you're able to launch the extension app?

**Comment by jtojnar** [comment]
I was seeing 

```
(gnome-shell-extension-prefs:27699): Gjs-WARNING **: 10:15:31.722: JS ERROR: Gio.DBusError: Error calling StartServiceByName for org.gnome.Shell.Extensions: Process org.gnome.Shell.Extensions exited with status 1
_injectToMethod/klass[method]@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:290:25
_makeProxyWrapper/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:267:17
vfunc_startup@resource:///org/gnome/Extensions/js/main.js:69:28
main@resource:///org/gnome/Extensions/js/main.js:499:23
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/nix/store/s5yirnjfa4mncn9ds4a4iplraj7nyxvl-gnome-shell-3.36.1/share/gnome-shell/org.gnome.Extensions:1:17

Gjs-Message: 10:15:31.722: JS WARNING: [resource:///org/gnome/Extensions/js/main.js 51]: reference to undefined property "_shellProxy"

(gnome-shell-extension-prefs:27699): Gjs-WARNING **: 10:15:31.722: JS ERROR: TypeError: this._shellProxy is undefined
vfunc_activate@resource:///org/gnome/Extensions/js/main.js:51:9
main@resource:///org/gnome/Extensions/js/main.js:499:23
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:5
@/nix/store/s5yirnjfa4mncn9ds4a4iplraj7nyxvl-gnome-shell-3.36.1/share/gnome-shell/org.gnome.Extensions:1:17
```

how did you deduce the missing typelib from that?

Weirdly `$out/share/dbus-1/services/org.gnome.Extensions.service` goes through bin/ which is wrapped.

GTK would not be enough since it depends on pango and other typelibs.

**Comment by hedning** [comment]
The tyeblib error pops up in the journal. It's indeed a bit weird that the wrapper doesn't work, since the service is launched through it as you note. But I guess there's a layer of indirection there somewhere, which causes the problem and results in the error being in the journal and not stderr.

**Comment by jtojnar** [comment]
Okay, so `org.gnome.Shell.Extensions` is a portal-like service in `$out/share/gnome-shell/org.gnome.Shell.Extensions` (https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/dbusServices/extensions/main.js) and `org.gnome.Extensions` is the app. The error was coming from the service but you were inserting the typelibs to the app, which failed because it depends on the service.

**Comment by jtojnar** [comment]
I wondered how does the `org.gnome.Shell.Notifications` work when it depends on `GLib` typelibs and it looks like `gjs` supports it even when it is not in `GI_TYPELIB_PATH`:

```ShellSession
$ nix run --ignore-environment -f . gjs -c gjs -c 'print(imports.gi.GLib)'
[object GIRepositoryNamespace]
```

**Comment by jtojnar** [comment]
Reverted the greeter logo patch and fixed the wrapping of the notification services but it is pretty ugly.

**Comment by hedning** [comment]
Looks good to me now, thanks for taking this on :)

**Review by jtojnar — APPROVED** [review]

**Pull request merged**